docker/centos7/Dockerfile

```diff
@@ -2,9 +2,14 @@ FROM centos:7 as build
 
 ARG DEVTOOLSET_VERSION=11
 
-WORKDIR /tmp
+ENV SCRATCH_DIR=/tmp/scratch
```

Comment:
For what ever reason this environment variable gets set empty, `rm -rf ${SCRATCH_DIR}/*` will cause an interesting result. What's the rationale behind this change? Was `/tmp` not good enough?

Reply:
This is just a question. I like this PR. Thank you!

Reply:
I do have some considerations here:

1. Ideally, when I build anything I should use `mkstemp` in order to avoid any pollution, i.e. isolate as much as possible. However, doing that is a lot more work.
2. I noticed that `/tmp` has some subdirectories that are not belonging to the steps, even though it seems deleting them will not cause any problem, not sure we should do this.
3. Using a custom scratch dir is a compromise to `mkstemp`. But it does isolate the build environment.
4. even `$SCRATCH_DIR` is '', the worst case is that the docker container, which is already an isolated environment, is not usable anymore, I feel this is not too much disastrous, as there is really nothing important/unrecoverable inside.

Please let me know if you have more concerns.